data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -546,6 +547,61 @@ private StateSyncCommitteesData getSyncCommitteesFromState(
             committeeIndices, spec.atEpoch(epoch).getConfig().getTargetCommitteeSize()));
   }
 
+  public SafeFuture<Optional<SyncCommitteeRewardData>> getSyncCommitteeRewardsFromBlockId(
+      final String blockId, final Set<String> validators) {
+    return getBlockAndMetaData(blockId)
+        .thenCompose(
+            result -> {
+              if (result.isEmpty() || result.get().getData().getBeaconBlock().isEmpty()) {
+                return SafeFuture.completedFuture(Optional.empty());
+              }
+              final BlockAndMetaData blockAndMetaData = result.get();
+              final BeaconBlock block = blockAndMetaData.getData().getBeaconBlock().get();
+              final SyncCommitteeRewardData data =
+                  new SyncCommitteeRewardData(
+                      blockAndMetaData.isExecutionOptimistic(), blockAndMetaData.isFinalized());
+
+              return combinedChainDataClient
+                  .getStateByBlockRoot(block.getRoot())
+                  .thenApply(
+                      maybeState ->
+                          getSyncCommitteeRewardData(validators, block, data, maybeState));
+            });
+  }
+
+  private Optional<SyncCommitteeRewardData> getSyncCommitteeRewardData(
+      Set<String> validators,
+      BeaconBlock block,
+      SyncCommitteeRewardData data,
+      Optional<tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState> maybeState) {
+    if (maybeState.isEmpty()) {
+      return Optional.empty();
+    }
```

Comment:
we go through handling optional inside this method just because the state coming in is optional, maybe its easier to just not call this method if the state is empty?
the upshot of that would be that then this method can just return SyncCommitteeRewardData
it may just mean using `maybeState.map()` where its used...